src/plugin/markers/index.js

```diff
@@ -0,0 +1,267 @@
+            color: params.color || DEFAULT_FILL_COLOR,
+            position: params.position || DEFAULT_POSITION
+        };
+
+        marker.el = this._createMarkerElement(marker);
+
+        this.wrapper.appendChild(marker.el);
+        this.markers.push(marker);
+        this._updateMarkerPositions();
+
+        return marker;
+    }
+
+    /**
+     * Remove a marker
+     *
+     * @param {number} index Index of the marker to remove
+     */
+    remove(index) {
+        let marker = this.markers[index];
+        if ( !marker ) return;
+
+        this.wrapper.removeChild(marker.el);
+        this.markers.splice(index, 1);
+    }
+
+    _createPointerSVG(color, position) {
+        const svgNS = "http://www.w3.org/2000/svg";
+
+        const el = document.createElementNS(svgNS, "svg");
+        const polygon = document.createElementNS(svgNS, "polygon");
+
+        el.setAttribute("viewBox", "0 0 40 80");
+
+        polygon.setAttribute("id", "polygon");
+        polygon.setAttribute("stroke", "#979797");
+        polygon.setAttribute("fill", color);
+        polygon.setAttribute("points", "20 0 40 30 40 80 0 80 0 30");
+        if ( position == "top" ) {
+            polygon.setAttribute("transform", "rotate(180, 20 40)");
+        }
+
+        el.appendChild(polygon);
+
+        this.style(el, {
+            width: this.markerWidth + "px",
+            height: this.markerHeight + "px",
+            "margin-right": "5px",
+            "z-index": 4
+        });
+        return el;
+    }
+
+    _createMarkerElement(marker) {
+        let label = marker.label;
+        let time = marker.time;
+
+        const el = document.createElement('marker');
+        el.className = "wavesurfer-marker";
+
+        this.style(el, {
+            position: "absolute",
+            height: "100%",
+            display: "flex",
+            "flex-direction": (marker.position == "top" ? "column-reverse" : "column")
+        });
+
+        const line = document.createElement('div');
+        this.style(line, {
+            "flex-grow": 1,
+            "margin-left": (this.markerWidth / 2 - 0.5) + "px",
+            background: "black",
+            width: "1px",
+            opacity: 0.1
+        });
+        el.appendChild(line);
+
+        const labelDiv = document.createElement('div');
+        const point = this._createPointerSVG(marker.color, marker.position);
+        labelDiv.appendChild(point);
+
+        if ( label ) {
+            const labelEl = document.createElement('span');
+            labelEl.innerText = label;
+            this.style(labelEl, {
+                "font-family": "monospace",
+                "font-size": "90%"
+            });
+            labelDiv.appendChild(labelEl);
+        }
+
+        this.style(labelDiv, {
+            display: "flex",
+            "align-items": "center",
+            cursor: "pointer"
+        });
+
+        el.appendChild(labelDiv);
+
+        labelDiv.addEventListener("click", e => {
```

Comment:
this event listener is never cleaned up is it?

Reply:
no but the div is removed from the DOM... would think that was adequate but not sure.

Reply:
add a console log in the click handler, destroy wavesurfer, create a new one and see if there are 2 click console logs printed..

Reply:
confirmed, the handler is removed.